talk/C++Course.tex

```diff
@@ -4,10 +4,22 @@
 
 \input{setup}
 
-\newboolean{onlybasics}
-\setboolean{onlybasics}{false}
+\newif\ifbasic % by default basic is false
+%\basictrue
 \IfFileExists{onlybasics.tex}{\input{onlybasics}}{}
 
+\ifbasic
+\excludecomment{advanced}
+\else
+\includecomment{advanced}
+\fi
+
+\newcommand\ifadvanced[1]{
+\ifbasic\else
+#1
+\fi
+}
+
```

Comment:
I don't know if it was just a misunderstanding, but I struggled a lot to understand what you're trying to to here. Before merging, I suggest to add some comments as to what's going on here.

Now, the biggest problem was to first read up on the comment package (which is only included at the very bottom of the PR -- not your fault), and then matching what I read there to what's going on here.
Now that I think I know what's going on, the biggest question is:
- Why are you declaring two ways of commenting in the advanced part?

Unless there's a latex reason I don't see, isn't it possible to do *everything* using `advanced` and `endadvanced`? E.g.
```latex
\begin{advanced}
  ...
\end{advanced}
```

Why is there a command `\ifadvanced` that's kind of doing the same thing?

Also, I found
```latex
\ifbasic\else
```
very confusing.
Maybe this is easier to read?
```latex
\ifbasic
%
\else
#1
\fi
```
or `\relax` etc. What do you think?

Reply:
PS:
I want to add that a really big advantage of the new method is that the `\input` duplication below goes away. I'm much in favour of this!

Reply:
I added a few comments to explain what I am trying to achieve.

> Why are you declaring two ways of commenting in the advanced part?

Because I thought having a command that just wraps the advanced part in parentheses would look nicer in some places. But you are correct, that it is just a duplication of functionality. I removed it now and handle everything using the `advanced` environment.